test/functional/p2p_sendrecon.py

```diff
@@ -0,0 +1,161 @@
+        peer.wait_for_verack()
+        assert not peer.sendrecon_msg_received
+
+        # Checks for the node *receiving* SENDRECON
+        self.log.info('valid SENDRECON')
+        peer = self.nodes[0].add_p2p_connection(SendReconSender(manual_verack=True), send_version=True, wait_for_verack=False)
+        peer.wait_until(lambda: peer.version_received)
+        peer.send_message(create_sendrecon_msg())
+        time.sleep(1)
+        assert peer.is_connected
+        self.log.info('second SENDRECON triggers a disconnect')
+        peer.send_message(create_sendrecon_msg())
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDRECON with initiator=responder=0 triggers a disconnect')
+        peer = self.nodes[0].add_p2p_connection(SendReconSender(manual_verack=True), send_version=True, wait_for_verack=False)
+        peer.wait_until(lambda: peer.version_received)
+        sendrecon_no_role = create_sendrecon_msg()
+        sendrecon_no_role.initiator = False
+        sendrecon_no_role.responder = False
+        peer.send_message(sendrecon_no_role)
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDRECON with initiator=1 and responder=0 from inbound triggers a disconnect')
+        peer = self.nodes[0].add_p2p_connection(SendReconSender(manual_verack=True), send_version=True, wait_for_verack=False)
+        peer.wait_until(lambda: peer.version_received)
+        sendrecon_wrong_role = create_sendrecon_msg()
+        sendrecon_wrong_role.initiator = False
+        sendrecon_wrong_role.responder = True
+        peer.send_message(sendrecon_wrong_role)
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDRECON with initiator=0 and responder=1 from outbound triggers a disconnect')
+        peer = self.nodes[0].add_outbound_p2p_connection(SendReconSender(), p2p_idx=3, connection_type="outbound-full-relay")
+        peer.wait_until(lambda: peer.version_received)
+        sendrecon_wrong_role = create_sendrecon_msg()
+        sendrecon_wrong_role.initiator = True
+        sendrecon_wrong_role.responder = False
+        peer.send_message(sendrecon_wrong_role)
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDRECON with version=0 triggers a disconnect')
+        peer = self.nodes[0].add_p2p_connection(SendReconSender(manual_verack=True), send_version=True, wait_for_verack=False)
+        peer.wait_until(lambda: peer.version_received)
+        sendrecon_low_version = create_sendrecon_msg()
+        sendrecon_low_version.version = 0
+        peer.send_message(sendrecon_low_version)
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDRECON after VERACK triggers a disconnect')
+        peer = self.nodes[0].add_p2p_connection(SendReconSender(manual_verack=True), send_version=True, wait_for_verack=False)
+        peer.wait_for_verack()
+        peer.send_message(msg_verack())
+        peer.send_message(create_sendrecon_msg())
+        peer.wait_for_disconnect()
```

Comment:
Do you have coverage for the case the peer does not send a WTXIDRELAY before VERACK, thus forgetting the reconciliation registration (L2855, in `src/net_processing.cpp`) ?